Writte reviews

Egoes98 opened the pull request:
Thsi branch has the methods and GUI fro Writting a review. This view should be implkemented in the showReservation once that part is done.  So when a user enters his reservation a button should appear that show him this GUI where we can write a review, press the summit button and the review is created in the DB.

Egoes98 committed:
WritteView

The view without the logic. Lacks localization and code should be cleaner.

Egoes98 committed:
View and methods

Added localization on the view and the methods and error messages. Created the methods needed to create new review.

Egoes98 committed:
WriteReview setters

Egoes98 committed:
Laclization and added to view system

Egoes98 committed:
More localization

Egoes98 committed:
Write review GUI and Methods working

Egoes98 committed:
Merge branch 'master' into writte-reviews

iker-barriocanal reviewed (dismissed):
- There's documentation missing.
- You have added `getUser` in `UserDAO`, is it needed in `IUserDAO`?

Egoes98 committed:
Documentation

iapellaniz commented on src/main/java/es/deusto/spq/client/gui/views/reviews/WriteReview.java:
```diff
@@ -0,0 +1,200 @@
+
+		//Sets correctly the frame.
+		frame.toFront();
+		frame.setVisible(true);
```
Missing `addDisposeEventHandler` method call at the end of the initialization.

iapellaniz commented on src/main/java/es/deusto/spq/server/data/dto/ReviewDTO.java:
```diff
@@ -31,25 +31,15 @@
-	 * @param user        the user DTO of the user that has posted the review.
 	 */
-	public ReviewDTO(String reviewID, String opinion, int score, Timestamp publishDate, HotelDTO hotel, UserDTO user) {
+	public ReviewDTO(String reviewID, String opinion, int score, Timestamp publishDate) {
```
If we don't provide the Hotel and User DTOs, what about providing their ids?

Egoes98 replied:
I remove it because it was producing erros and they are only necesary for getting them from the DB, but i will try to find a solution and add the ids at least.

Egoes98 committed:
Added addDisposeHandler()

Egoes98 committed:
Check Changed

I havent been able to get users from inside the reveiw. So i have changed the way the check to see if user has alredy posted a review about that hotel works. Now the ReveiwID is  mix of the userID that writtes and the hotelID the hotel the review is for. In this way if a reviewID is the same it means user alredy has a review on that hotel.

Egoes98 committed:
Merge branch 'master' into writte-reviews
